ocaml/typing/typemod.ml

```diff
@@ -3298,15 +3362,14 @@ and type_structure ?(toplevel = None) ~expected_sig funct_body anchor env sstr =
         add_expected_rec_modules_to_subst expected_sig idents subst
     | Pstr_modtype pmtd ->
         let context =
-          let get_modtype_decl = function
-            | Sig_modtype (ident, decl, _)
-                when Ident.name ident = pmtd.pmtd_name.txt ->
-                  Some (Subst.modtype_declaration Keep subst decl)
-            | _ -> None
-          in
           match expected_sig with
           | None -> In_structure None
-          | Some sg -> In_structure (List.find_map get_modtype_decl sg)
+          | Some expected_sig ->
+            begin match Sig_map.find_module_type pmtd.pmtd_name.txt expected_sig with
+              | None -> In_structure None
+              | Some (_, decl) ->
+                  In_structure (Some (Subst.modtype_declaration Keep subst decl))
+            end
```

Comment:
minor formatting nit: insert two extra space indentation

Reply:
Also, might be worthwhile commenting at the definition of `In_structure` that the expected modtype decl has gone through substitution.

Reply:
Sorry, I'm not sure what you mean about the spacing - where should I add the spaces? The file seems to use a number of conflicting conventions.

Reply:
I meant to add two extra space indendation to the quoted five lines of code.

You are right that the file is not consistent in this regards - and I'm guilty for some of that. Regardless, new code should try to follow the rules.

Reply:
Ah I see now - thanks!